arch/arm/src/cxd56xx/cxd56_serial.c

```diff
@@ -1001,9 +1001,13 @@ static void up_txint(FAR struct uart_dev_s *dev, bool enable)
        * interrupts disabled (note this may recurse).
        */
 
+#  ifdef CONFIG_SMP
       spin_unlock_irqrestore(&priv->lock, flags);
```

Comment:
what about SMP case? The buffer will not be corrupted?

Reply:
@pkarashchenko 
I've been testing spresense:wifi_smp and spresense:rndis_smp but I've never seen the corruption.

Reply:
@pkarashchenko 
Yeah, but I've never seen such corruption in SMP.

Reply:
@pkarashchenko 
If we find such corruption in SMP mode in the future, I would revert the changes. (i.e. use critical section API instead of spinlock API for cxd56_serial.c)